codec/address_test.go

```diff
@@ -39,3 +39,87 @@ func TestAddressJSON(t *testing.T) {
 	require.NoError(json.Unmarshal(addrJSONBytes, &parsedAddr))
 	require.Equal(addr, parsedAddr)
 }
+
+func TestAddressStringIdentity(t *testing.T) {
+	require := require.New(t)
+	typeID := byte(0)
+	addrID := ids.GenerateTestID()
+
+	addr := CreateAddress(typeID, addrID)
+
+	originalAddr, err := StringToAddress(addr.String())
+	require.NoError(err)
+	require.Equal(addr, originalAddr)
+}
+
+func TestAddressToString(t *testing.T) {
+	t.Run("zeroAddressChecksum", func(t *testing.T) {
+		require := require.New(t)
+		expectedStr := "0x000000000000000000000000000000000000000000000000000000000000000000a7396ce9"
+		require.Equal(expectedStr, EmptyAddress.String())
+	})
+
+	t.Run("arbitraryAddressChecksum", func(t *testing.T) {
+		require := require.New(t)
+		expectedStr := "0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20a10df6ab"
+		addr := CreateAddress(
+			byte(0),
+			[32]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32},
+		)
+		require.Equal(expectedStr, addr.String())
+	})
```

Comment:
If there's no shared structure between these tests, can we simply break them into two different tests rather than use `t.Run` ?

Reply:
Done